src/modules/assets/assets.entity.ts

```diff
@@ -0,0 +1,37 @@
+import { Field, Int, ObjectType } from '@nestjs/graphql';
+import { Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
+import { DIDEntity } from '../did/did.entity';
+
+@Entity()
+@ObjectType()
+export class Asset {
+  static create(data: Partial<Asset>) {
+    const entity = new Asset();
+    Object.assign(entity, data);
+    return entity;
+  }
+  @PrimaryColumn({ type: 'text' })
+  @Field(() => Int)
+  id: string;
+
+  @Column()
+  @Field()
+  owner: string;
+
+  @Column({ nullable: true })
+  @Field({ nullable: true })
+  offeredTo?: string;
+
+  @Column()
+  @Field()
+  timestamp: string;
+
+  @OneToOne(
+    () => DIDEntity,
+    did => did.id,
+    { eager: true },
+  )
+  @JoinColumn()
+  @Field(() => DIDEntity)
+  document: DIDEntity;
```

Comment:
I think this approach of having the Asset being composed fo the DIDEntity 👍 
So in postgres, Assets and DID(Documents) will each be a separate table?

Reply:
Yes, Asset and DIDDoc are seperate tables in postgres and asset has onetoone relation with didDoc :)